docs/examples/Ch-ExamplesAddingMQTTDevice.rst

```diff
@@ -0,0 +1,216 @@
+###############################
+MQTT - Adding a Device to EdgeX
+###############################
+
+In this example, we use the simulator instead of a real device. This provides a straight-forward way to test the device-mqtt features.
+
+.. image:: MQTTdiagram.png
+    :scale: 50%
+    :alt: MQTT Setup
+
+Environment
+-----------
+
+You can use any operating system that can install docker and docker-compose. In this example, we use Photon OS to delpoy EdgeX using docker. The system requirements can be found at https://docs.edgexfoundry.org/Ch-GettingStartedUsers.html#what-you-need.
+
+.. image:: PhotonOS.png
+    :scale: 50%
+    :alt: Photon Operating System
+
+.. image:: PhotonOSversion.png
+    :scale: 50%
+    :alt: Version Information
+
+
+Set up the Device mqtt-go Device Service, as described in https://docs.edgexfoundry.org/Ch-ExamplesMQTTDeviceService.html
+
+MQTT Broker
+-----------
+
+EMQ is a MQTT broker that is easy to configure and use. You can download it from http://emqtt.io. It’s fully open source and licensed under Apache Version 2.0. 
+
+Run using the following docker command::
+
+    docker run -d --rm --name emqx -p 18083:18083 -p 1883:1883 emqx/emqx
+
+After the MQTT broker has started, login to the dashboard using the default account details (name = admin, password = public)
+
+.. image:: MQTTBroker.png
+    :scale: 50%
+    :alt: MQTT Broker
+
+MQTT Device (Simulator)
+-----------------------
+
+The device-mqtt source code provides a mock device that can send async values and handle commands. You must implement these two functions when using the device-mqtt service as described below.
+
+.. image:: MQTTDeviceService.png
+    :scale: 50%
+    :alt: MQTT Device Service
+
+To set up the service:
+
+1. Download the source code::
+
+    git clone git@github.com:edgexfoundry-holding/device-mqtt-go.git
+
+2. Modify the broker connection
+
+   The broker connection information must be modified to meet our requirement. 
+
+    .. image:: devicegoSetup.png
+        :scale: 50%
+        :alt: device.go Set Up
+
+3. Run the device simulator
+
+   Run the following command in your terminal::
+ 
+    go run ./mock/device.go
+
+Set Up
+------
+
+The following sections describe how to complete the set up.
+
+Set Up Device Profile
+=====================
+
+The DeviceProfile defines the device's values and operation method, which can be Read or Write. 
+
+    .. image:: deviceprofile.png
+        :scale: 50%
+        :alt: Device profile
+
+Set Up Device Service Configuration
+===================================
+
+Use this configuration file to define devices and schedule jobs. The device-mqtt generates a relative instance on startup.
```

Comment:
You will need to add your configuration.toml file to this directory, and name it EdgeX_ExampleMQTT_configuration.toml

There is another configuration.toml already used elsewhere in the documentation, so the download link is picking that up. To avoid this, you have to add your configuration.toml with a unique name here.

Reply:
Added file and updated link